### PR TITLE
Add Joke Services and Tests

### LINKED ISSUE #6: Add Joke Service and Tests
# Acceptance Criteria:

- [ ] There is <tt>JokeQueryService.java</tt> file in the `src/main/java/edu/ucsb/cs156/spring/backend/services` directory.
- [ ] <tt>JokeQueryService.java</tt>  implements a class that make a call to the appropriate API
      and returns the results.
- [ ] There is  <tt>JokeQueryServiceTests.java</tt> file in the `src/test/java/edu/ucsb/cs156/spring/backend/services` directory.
- [ ] The tests in  <tt>JokeQueryServiceTests.java</tt> provide 100% coverage of  <tt>JokeQueryService.java</tt> for both jacoco and pitest.

# Details

See the assignment description at https://ucsb-cs156.github.io/s24/lab/team01 for details.

### PR DESCRIPTION
**Pull request by vaaniarora**
In this PR, we add a service that wraps Joke API from https://v2.jokeapi.dev/joke/{category}?amount={numJokes}.

Closes #6

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/spring/backenddemo/services/JokeQueryService.java [hunk 1]
```diff
@@ -1,23 +1,48 @@
...
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+
 
+@Slf4j
 @Service
 public class JokeQueryService {
+    ObjectMapper mapper = new ObjectMapper();
```
Comment (Jumshim) [inline src/main/java/edu/ucsb/cs156/spring/backenddemo/services/JokeQueryService.java]: 🤓  is this unnecessary? I realized that it's not used anywhere after this line. Maybe the import statement can be removed too

File: src/main/java/edu/ucsb/cs156/spring/backenddemo/services/JokeQueryService.java [hunk 2]
```diff
@@ -1,23 +1,48 @@
...
     private final RestTemplate restTemplate;
 
     public JokeQueryService(RestTemplateBuilder restTemplateBuilder) {
         restTemplate = restTemplateBuilder.build();
     }
 
-    public static final String ENDPOINT = "";
+    public static final String ENDPOINT = "https://v2.jokeapi.dev/joke/{category}?amount={numJokes}";
 
     public String getJSON(String category, int numJokes) throws HttpClientErrorException {
```
Comment (Jumshim) [inline src/main/java/edu/ucsb/cs156/spring/backenddemo/services/JokeQueryService.java:32]: Not the change, but just noting: is `numJokes` int or string? Why was it given as an `int` but the example Swagger UI website lists it as a `string`

http://team01.dokku-00.cs.ucsb.edu/swagger-ui/index.html#/joke-controller/getJokesUsingGET

**Review by Jumshim — COMMENTED** [review]
Some things can be taken out to reduce unused sections of code (I don't know if it is/isn't used), but once that's done, LGTM

**Comment by Jumshim** [comment]
LGTM!

**Review by Jumshim — APPROVED** [review]
LGTM

**Review by rainakakani21 — APPROVED** [review]
lgtm!

**Review by niyati-m — APPROVED** [review]
LGTM

**Review by emilythai — APPROVED** [review]
LGTM!

**Review by charlene-patenaude — APPROVED** [review]
lgtm :)

**Pull request merged**